Vungle/src/main/java/com/mopub/nativeads/VungleNative.java

```diff
@@ -0,0 +1,342 @@
+        assembleChildViewsWithLimit(view, clickableViews, 10);
+
+        if (clickableViews.size() == 1) {
+            nativeAd.registerViewForInteraction(view, mediaView, adIconView);
+        } else {
+            nativeAd.registerViewForInteraction(view, mediaView, adIconView, clickableViews);
+        }
+    }
+
+    private static void assembleChildViewsWithLimit(final View view,
+                                                    final List<View> clickableViews, final int limit) {
+        if (view == null) {
+            MoPubLog.log(getAdNetworkId(), CUSTOM, "View given is null. Ignoring");
+            return;
+        }
+
+        if (limit <= 0) {
+            MoPubLog.log(getAdNetworkId(), CUSTOM, "Depth limit reached; adding this view " +
+                    "regardless of its type.");
+            clickableViews.add(view);
+            return;
+        }
+
+        if (view instanceof ViewGroup && ((ViewGroup) view).getChildCount() > 0) {
+            final ViewGroup vg = (ViewGroup) view;
+            for (int i = 0; i < vg.getChildCount(); i++) {
+                assembleChildViewsWithLimit(vg.getChildAt(i), clickableViews, limit - 1);
+            }
+            return;
+        }
+
+        clickableViews.add(view);
+    }
+
+    @NonNull
+    private static String getAdNetworkId() {
+        return mPlacementId != null ? mPlacementId : "";
+    }
+
+    static class VungleNativeAd extends BaseNativeAd implements NativeAdListener {
+
+        private final NativeAd nativeAd;
+        private final CustomEventNativeListener customEventNativeListener;
+
+        private final Map<String, Object> mExtras;
+
+        VungleNativeAd(NativeAd nativeAd, CustomEventNativeListener customEventNativeListener) {
+            this.nativeAd = nativeAd;
+            this.customEventNativeListener = customEventNativeListener;
+            mExtras = new HashMap<String, Object>();
+        }
+
+        @Override
+        public void prepare(@NonNull View view) {
+            // no-op
+        }
+
+        @Override
+        public void clear(@NonNull View view) {
+            // no-op
+        }
+
+        @Override
+        public void destroy() {
+            nativeAd.destroy();
+        }
+
+        void loadAd() {
+            nativeAd.loadAd(new NativeAdConfig(), this);
+            MoPubLog.log(getAdNetworkId(), LOAD_ATTEMPTED, ADAPTER_NAME);
+        }
+
+        Map<String, Object> getExtras() {
+            return new HashMap<>(mExtras);
+        }
+
+        void addExtra(final String key, final Object value) {
+            if (!Preconditions.NoThrow.checkNotNull(key, "addExtra key is not allowed " +
+                    "to be null")) {
+                return;
+            }
+            mExtras.put(key, value);
+        }
+
+        /**
+         * Returns the String corresponding to the ad's title.
+         */
+        final public String getTitle() {
+            return nativeAd.getAdHeadline();
```

Comment:
Here's the confusing API. @felixzhang0703 @ashinagawa

Reply:
I suggest that we rename this api to nativeAd.getTitle() from sdk side. And in integration document, we can add description to this API. 
cc @HuFangshuai @ashinagawa